lib/apq/document_provider.ex

```diff
@@ -10,15 +10,16 @@ defmodule Apq.DocumentProvider do
   defmodule ApqExample.Apq do
     use Apq.DocumentProvider,
       cache_provider: ApqExample.Cache,
-      max_query_size: 16384 #default
+      max_query_size: 16384 # default
 
   end
   ```
 
   #### Options
 
   - `:cache_provider` -- Module responsible for cache retrieval and placement. The cache provider needs to follow the `Apq.CacheProvider` behaviour.
-  - `:max_query_size` -- (Optional) Maximum number of bytes of the graphql query document. Defaults to 16384 bytes (16kb)
+  - `:max_query_size` -- (Optional) Maximum number of bytes of the graphql query document. Defaults to 16384 bytes (16kb).
+  - `:json_codec` -- (Optional) Only required if using GET for APQ's hashed queries.  Must respond to `decode!/1`.
```

Comment:
I couldn't find a way to use the `json_codec` that might otherwise be passed to `Absinthe.Plug.init`, so it has to be passed to `use Apq.DocumentProvider`.  I'm open to suggestions if there's a way to use the existing `json_codec`.

Reply:
I agree, it doesn't look like there is a way to access the options passed to Abinsthe.Plug within the document provider unfortunately.